### DISCUSSION TITLE
Bally Williams Ball Trough v12 opto + cobrapin Does it work?

### DISCUSSION THREAD

**Discussion by agrisdul** [missionpinball/missionpinball#49] [Q&A] [answered]
Dear community, seams I run into dead end. 

I Ordered ball trough with 12v opto. "5768-13660-01" #A-17982. This ball trough is not in the list on [missionpinball](https://docs.missionpinball.org/en/latest/mechs/troughs/modern_opto.html)   reading this [How to configure opto switches](https://docs.missionpinball.org/en/latest/mechs/switches/optos.html) can't understand how to manage this.

Was read this group discussion and this is for spooky ball trough [Google Group opto discussion](https://groups.google.com/g/mpf-users/c/Cn_2ZoLyBpY)

1. question: 
How I can get 12v on cobrapin? I have just 5v and 48v power supply.  

2. Does wiring will be the same as here [Opto's with Cobrapin disussion](https://groups.google.com/g/mpf-users/c/Cn_2ZoLyBpY)

How should I proceed? Who had similar experience

**Comment by agrisdul** [comment]
Ok, seams found some clue. This could work?

_J10:
Power input for Neopixels, most likely 5V, but if you use 12V Neopixels you need to provide 12V power here. Power input is used for both Neopixel chains._ [neopixel support](https://docs.missionpinball.org/en/latest/hardware/opp/cobrapin/index.html#neopixel-support)

**Comment by Pucksnok** [comment]
I would say since you link the transmitter you just feed it with 12V directly from a 12V power supply, no need to go through a cobrapin board for power.

Not sure what receiver you use, could need some fixing there but it's depend, probably not needed. Can you link it?

**Reply by agrisdul** [reply]
Thanks. 
I have this one Blue in one side: https://www.marcospecialties.com/pinball-parts/A-17981
And this one green on another side: https://www.marcospecialties.com/pinball-parts/A-17982 

Just thinking it will not hurt Cobrapin? if I will connect them to switches. 

So Your idea is to power both sides with 12v from separate supply.  
I have contacts on blue side like: R5 , R4, R6, R7, R3, R1, R2, COL, 12v, GND  So seams it is receivers, that I have to connect to the Cobrapin as a switches. Don't know what means COL.  

IMg: https://imgur.com/a/CjGbuan

On the green side i have just 2x12v and 2xGND



Don't want to buy another trough with mechanical switches :(

**Reply by Pucksnok** [reply]
The transmitter PCB is safe to use with 12V but it's way high to use for the receiver PCB if you connect it to your cobra board (which you need...).
I'm not sure what is needed to change on the receiver without a schematic on the pcb but someone here may/should be able to tell. You probably just need to change some resistors...

**Reply by agrisdul** [reply]
Thanks for your input, seams need to order another trough. I will check what else can be done

**Comment by ralfit0** [comment]
Receiver schematic can be found e.g. on page 3-16 from www.progettosnaps.net/manuals/pdf/dm_lx4.pdf
I am no good in giving solid advice there. You *may* be fine running the board at a lower voltage than the original 12VDC; or the 12VDC *may* not be the output on the switch side towards the Cobra board, but I could not tell.

**Comment by cobra18t** [comment]
The transmitter is definitely fine to power from 12V since it has no connection to the CobraPin board. 

The receiver looks promising, but let me do some more checking/testing here. I have a Demo Man that I just brought to my house. The row outputs that go to the CobraPin are diode protected so that it won't overvoltage the CobraPin. I need to review the column input more, though. I will come up with a recommendation for you soon.

**Reply by cobra18t** [reply]
Is R14 missing from your receiver board? It is on this Demo Man receiver. If you populate that with a 33k ohm resistor, that should bring the column voltage down to a safe level. Don't do it yet, let me try it on mine first and I will let you know.

**Reply by cobra18t** [reply]
I just tried the 33k ohm resistor in R14 and it worked. The column voltage was at a safe level and the trough still functioned. Let me know if your R14 is also unpopulated.

**Reply by agrisdul** [reply]
Thank You guys for fast and detailed help. 

@ralfit0 thank You for documentation.
@cobra18t Indeed I have empty R14. So I have to add there 33k ohm resistor. (does is important what side is what for resistor)

Can I please summary up:

1. Add resistor in R14
2. Then power Green board (transmitter) from external 12v power source.
3. And does receiver side (blue board) also has to be powered from external 12v power source
4. R1-R7 pins should be connected to cobrapin switches
5. Does cobrapin switch GND should be connected with receiver GND same where. 

Sorry for the question, at evening will start testing :D

**Reply by agrisdul** [reply]
Schematic should be the same as here: [Google group img](https://groups.google.com/group/mpf-users/attach/189834d6dc4c4/wire2_BallTroughOptos.jpg?part=0.1&view=1) just red line will be external 12v power

**Reply by Pucksnok** [reply]
Resistors doesn't have polarity so it doesn't matter which direction you solder them.

Use 12V for both boards.

**Reply by cobra18t** [reply]
With the R14 mod, both the transmitter and receiver can be powered from an external 12V supply. The GND from the CobraPin and that external supply must be connected. Assuming you are running a switch matrix, the row pins from the receiver go to J4 (SW MATRIX IN) on the CobraPin and the column pin on the receiver goes to J5 (SW MATRIX OUT) on the CobraPin.

**Reply by agrisdul** [reply]
Hi, @cobra18t  I will not use switch matrix. I have programmed as direct switches . So every R has to be own switch?

**Reply by cobra18t** [reply]
@agrisdul if you are doing direct switches, you don't even need the R14 mod. Sorry I did not ask that sooner. Just connect the column pin to GND. Then, yes, each row pin becomes its own direct switch to be connected to the CobraPin.

**Reply by agrisdul** [reply]
My bad I didn’t explained all environment.

you programmed this for me as all direct switches long time ago. 
So everything is ok can work out 

I should connect transmitter and receiver to external 12v source. 
connect Col to GND (cobrapin) and R1-R7 to switches like normal switch. 
I didn’t had chance to experiment. I will try and le you know

**Comment by agrisdul** [comment, marked as answer]
Thank You guys for your support. 

So This could be the wiring for Data east opto trough with cobrapin using direct switch?  Does Ground should be common?  

green board - transmitter 
blue board - receiver
![image](https://i.imgur.com/ZmlhXhQ.jpg)

Merry Christmas to everyone

**Reply by cobra18t** [reply]
Yes, both GNDs should be connected. That connection scheme should work just fine. There are potential optimizations, but nothing significant.

**Reply by agrisdul** [reply]
Ok, finally got on wiring and got nothing.. I mean "transmitter led works- are on" (checked with phone camera) But in cobra pin all tough switches are green (NC) without any changes if I put ball in the trough.   Switches are not reacting, does cobra gnd should go only to COL like i have?

Notice that R4,R5 and other are getting hot when touching, maybe to much power? (12v, 4.2A) Or this heat is ok. 

This is wiring what I have used: 
![image](https://i.imgur.com/cEraWLu.png) source https://www.progettosnaps.net/manuals/pdf/dm_lx4.pdf

**Reply by cobra18t** [reply]
Is there continuity between "Cobra switch GND" and "GND - power supply?"

Those resistors are dissipating ~370mW each, so they are going to get warm.

**Reply by agrisdul** [reply]
Hi. 

No they are not connected together. How should I connect them?

**Reply by agrisdul** [reply]
I used this schematic, but didn’t connected 48v to board, just tested switches 

![image](https://user-images.githubusercontent.com/4344783/210496902-e0695a31-e038-4649-b7d4-70bff1fafcc8.jpeg)

**Reply by cobra18t** [reply]
Connect the 12V Supply GND to any of the GND connections on the CobraPin. In your diagram, you show it connected via the 48V connector. You can connect it somewhere else if you don't want to make that connection.

**Reply by agrisdul** [reply]
Thanks @cobra18t i will try today after work. 

What about switch input (where goes COL and buttons) GND I can connect there too?  I haven’t thinking about it they before

**Reply by agrisdul** [reply]
@cobra18t Thank You connected like in schema and it works! Thank You very much Guys for support. topic closed.

**Reply by cobra18t** [reply]
Glad you got it working. Be sure to post your project here: https://missionpinball.org/projects.html
